### PR TITLE
really drop python <=3.7 support

### PR DESCRIPTION
**Pull request by kloczek**
Filetrer all cod over `pyupgrade --py38-plus`.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -25,7 +25,6 @@
 copyright = f"2013-{datetime.date.today().year}, Aymeric Augustin and contributors"
 author = "Aymeric Augustin"
 
-from websockets.version import tag as version, version as release
```
Comment (aaugustin) [inline docs/conf.py:28]: That will break generating the docs.

File: example/django/notifications.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import django
 import websockets
 
+
```
Comment (aaugustin) [inline example/django/notifications.py:10]: Please avoid gratuitous formatting changes. Also, probably the autoformatter built into the project would revert it at the next commit.

File: example/echo.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
 #!/usr/bin/env python
 
 import asyncio
+
 from websockets.server import serve
 
+
```
Comment (aaugustin) [inline example/echo.py:7]: Again, gratuitous change that will make the docs look worse.

**Review by aaugustin — CHANGES_REQUESTED** [review]
Some of the changes are valid — e.g. unnecessary imports — but there's also many incorrect changes. Unfortunately it will probably be faster for me to redo the changes than to review and guide you over GitHub comments :-/ I gave up reviewing after a few files because the signal / noise ratio wasn't great.

**Comment by aaugustin** [comment]
I think I'm going to redo individual changes — again, faster to redo than review — which will eventually lead to the result that you're aiming for. Thank you for the sugestion! That's a good opportunity to clean up.

**Comment by kloczek** [comment]
OK so just cannel this PR and do that your way.
If you see some code wrongly filtered please discuss that with `pyupgrade` maintainer.
FYI I've filtered `websockets` code up to --py310-plus and it passes test suite and I don't see any issues on interaction with other modules. code.

**Comment by aaugustin** [comment]
Union types were introduced in Python 3.10. See https://peps.python.org/pep-0604/. As a consequence, this change isn't compatible with Python 3.8.

```
$ python3.8
Python 3.8.16 (default, Apr  2 2023, 07:49:54)
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> def foo(bar=str | None): pass
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
>>>
```

```
$ python3.10
Python 3.10.10 (main, Apr  2 2023, 07:49:30) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> def foo(bar=str | None): pass
...
>>>
```

**Comment by aaugustin** [comment]
Oops. Actually not a problem thanks to `from __future__ import annotations`. I had missed that.

**Comment by aaugustin** [comment]
Need to check that the docs still generate properly.

**Comment by aaugustin** [comment]
As of f45286b all the good changes from this PR are in the main branch.

**Pull request closed without merging**